### PR TITLE
EG-651, Added aria-describedby to fieldset wrapping radios referencing hint and error ids when present

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/views/components/question.scala.html [hunk 1]
```diff
@@ -20,10 +20,19 @@
 
 @(q: Question, name: String, form: Form[_])(implicit request: Request[_], messages: Messages)
 
+@ariaDescribedBy(s: String) = { aria-describedby="@s" }
+
+@optionalAria(a: Question) = @{
+  (q.hint.map(_ => "question-hint")::q.errorMsgs.headOption.map(err => s"${err.id}-error")::Nil).flatten match {
```
Comment (woollett-r) [inline app/views/components/question.scala.html:26]: That is an inspired line. If comments worked I would suggest you add one to explain what is going on!

Comment (cmrhmrc) [reply app/views/components/question.scala.html:26]: ;)

File: test/views/components/QuestionSpec.scala [hunk 2]
```diff
@@ -158,23 +158,28 @@ class QuestionSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
...
 
     "question without body should render hint within a span within fieldset" in new Test {
       val doc = asDocument(components.question(questionWithHintAndNoBody, "test", formProvider("test"))(fakeRequest, messages))
-      val legend = doc.getElementsByTag("fieldset").first
-      Option(legend.getElementsByTag("span").first).fold(fail("Missing hint span within fieldset")) { span =>
-        val attrs = elementAttrs(span)
-        attrs("id") shouldBe "question-hint"
-        attrs("class").contains("govuk-hint") shouldBe true
-        span.text shouldBe questionHint(0)
+      val fieldset = doc.getElementsByTag("fieldset").first
```
Comment (woollett-r) [inline test/views/components/QuestionSpec.scala:175]: Have you missed the aria-describedby test here?

File: test/views/components/QuestionSpec.scala [hunk 3]
```diff
@@ -259,23 +264,28 @@ class QuestionSpec extends WordSpec with Matchers with GuiceOneAppPerSuite {
...
     "question without body should render hint within a span within fieldset" in new WelshTest {
       val doc = asDocument(components.question(questionWithHintAndNoBody, "test", formProvider("test"))(fakeRequest, messages))
-      val legend = doc.getElementsByTag("fieldset").first
-      Option(legend.getElementsByTag("span").first).fold(fail("Missing hint span within fieldset")) { span =>
-        val attrs = elementAttrs(span)
-        attrs("id") shouldBe "question-hint"
-        attrs("class").contains("govuk-hint") shouldBe true
-        span.text shouldBe questionHint(1)
+      val fieldset = doc.getElementsByTag("fieldset").first
+      Option(fieldset).fold(fail("Missing fieldset")){ fset =>
```
Comment (woollett-r) [inline test/views/components/QuestionSpec.scala:282]: As above

File: test/views/components/QuestionSpec.scala (hunk 2 above)
Comment (cmrhmrc) [reply test/views/components/QuestionSpec.scala:175]: will add

File: test/views/components/QuestionSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/views/components/QuestionSpec.scala:282]: will add

**Pull request merged**